feat: add run-many

mbriggs opened the pull request:
- want some feedback on this, commented out affected.spec.ts, that still needs to be implemented
 - _think_ it would be a good idea to make `--projects` and `--all` to be exclusive/required
 - add a new command (run-many) that basically is the same as affected, but allows you to specify projects to run
 - re-structure affected.ts so that the bulk of the functionality can be shared

vsavkin commented on packages/workspace/src/command-line/nx-commands.ts:
```diff
@@ -286,6 +293,54 @@ function withAffectedOptions(yargs: yargs.Argv): yargs.Argv {
+    .option('verbose', {
+      describe: 'Print additional error stack trace on failure'
+    })
+    .option('plain', {
```
I don't think `affected:apps` and `affected:libs` will work with run-many. And this flag is only used by the two commands.

mbriggs commented:
- added --all, made it mutually exclusive with --projects, but required
 - yargs has super weird array input, changed --projects to be comma delimited
 - show warnings when asking to run a project that doesn't have configuration for target
 - show errors when asking to run a project that doesn't exist in the project
 - change the default reporter copy to not say "affected" (think whats there works fine in both cases)
 - rebase on to latest master (includes some bug fixes to affected.ts)

Still need to add tests

vsavkin commented on packages/workspace/src/command-line/run-tasks/run-command.ts:
```diff
@@ -0,0 +1,205 @@
+  );
+
+  const tasksMap: TasksMap = {};
+  projects.forEach(project => {
```
TaskMap is a set of all possible tasks, so it cannot be derived from the list of projects you are currently running.

vsavkin replied:
In the current version is done like this:

Object.entries(affectedMetadata.dependencyGraph.projects)

So we generate a task map from the list of all possible projects

vsavkin commented on packages/workspace/src/command-line/run-tasks/run-command.ts:
```diff
@@ -0,0 +1,205 @@
+type RunArgs = yargs.Arguments & ReporterArgs;
+
+export function runCommand<T extends RunArgs>(
+  projects: ProjectNode[],
```
I'd rename it to indicate that this isn't list of all possible projects but only the projects are running tasks against this time.

vsavkin commented on packages/workspace/src/command-line/run-tasks/run-many.ts:
```diff
@@ -0,0 +1,82 @@
+  projects.forEach(p => (map[p.name] = p));
+
+  return {
+    projects: map,
```
DependencyGraph must contain all projects. This map is only the projects you are running tasks against.

vsavkin commented on packages/workspace/src/command-line/run-tasks/run-many.ts:
```diff
@@ -0,0 +1,82 @@
+  return args;
+}
+
+function projectListToGraph(projects: ProjectNode[]): DependencyGraph {
```
DependencyGraph describes the workspace. It must contain all projects, all dependenciesd and real roots. This function doesn't do this.

vsavkin commented on packages/workspace/src/command-line/run-tasks/run-command.ts:
```diff
@@ -0,0 +1,205 @@
+
+export function runCommand<T extends RunArgs>(
+  projects: ProjectNode[],
+  projectGraph: DependencyGraph,
```
why not call it dependencyGraph?
